### PR TITLE
Update testing-your-github-pages-site-locally-with-jekyll.md

### LINKED ISSUE #34181: Add guidance for local Jekyll preview with custom baseurl
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll

### What part(s) of the article would you like to see updated?

The "Testing your site locally" section needs updating. Add information about running Jekyll locally when `_config.yml` has a baseurl set to the GitHub repository's link. This change supports users who encounter local preview issues due to baseurl conflicts. The expected outcome is clearer guidance on using `bundle exec jekyll serve --baseurl=""` to run Jekyll locally without modifying `_config.yml`, improving the local development experience for GitHub Pages sites.

### Additional information

This issue affects users who set their baseurl in `_config.yml` to match their GitHub Pages URL structure, which is a common practice. The problem occurs consistently for these users when trying to preview their site locally. While the exact number of affected users is unknown, it's a recurring issue for developers working with GitHub Pages. No screenshots seemed necessary, as the change involves adding a command and explanation to the documentation.

### PR DESCRIPTION
**Pull request by g-nitin**
### Why:

Some users set their `baseurl` in `_config.yml` to match their GitHub Pages URL structure. This can cause problems when previewing the site locally. The added information provides a simple solution without requiring changes to the configuration file.

Closes: #34181

### What's being changed (if available, include any code snippets, screenshots, or gifs):

This PR adds information about running Jekyll locally when the `_config.yml` file has a `baseurl` set to the GitHub repository's link. The changes include:

1. A new command `bundle exec jekyll serve --baseurl=""` to run Jekyll locally without modifying the `_config.yml` file.
2. A brief explanation of when and why to use this command.

These additions will help users who encounter issues with local preview due to `baseurl` conflicts, improving the local development experience for GitHub Pages sites.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@g-nitin Thanks so much for opening a PR! I'll get this triaged for review ✨

File: content/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll.md [hunk 1]
```diff
@@ -51,9 +51,17 @@ Before you can use Jekyll to test a site, you must:
 
    {% note %}
 
-   **Note:** If you've installed Ruby 3.0 or later (which you may have if you installed the default version via Homebrew), you might get an error at this step. That's because these versions of Ruby no longer come with `webrick` installed.
+   **Notes:**
+   - If you've installed Ruby 3.0 or later (which you may have if you installed the default version via Homebrew), you might get an error at this step. That's because these versions of Ruby no longer come with `webrick` installed.
+
+     To fix the error, try running `bundle add webrick`, then re-running `bundle exec jekyll serve`.
+
+   - If your `_config.yml` file's `baseurl` field contains your GitHub repository's link, you can use the following command to run the site locally without changing the `baseurl` in the `_config.yml` file:
```
Comment (tcbyrd) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll.md:59]: Great tip! I have a minor suggestion on wording to help connect why you'd want to do this when running locally:

```suggestion
   - If your `_config.yml` file's `baseurl` field contains your GitHub repository's link, you can use the following command when building locally to ignore that value and serve the site on `localhost:4000/`:
```

Comment (g-nitin) [reply content/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll.md:59]: That's a great suggestion @tcbyrd 
It does help clarify how and why the command is useful.

**Review by nguyenalex836 — DISMISSED** [review]

File: content/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll.md [hunk 2]
```diff
@@ -51,9 +51,17 @@ Before you can use Jekyll to test a site, you must:
...
-   **Note:** If you've installed Ruby 3.0 or later (which you may have if you installed the default version via Homebrew), you might get an error at this step. That's because these versions of Ruby no longer come with `webrick` installed.
+   **Notes:**
+   - If you've installed Ruby 3.0 or later (which you may have if you installed the default version via Homebrew), you might get an error at this step. That's because these versions of Ruby no longer come with `webrick` installed.
+
+     To fix the error, try running `bundle add webrick`, then re-running `bundle exec jekyll serve`.
+
+   - If your `_config.yml` file's `baseurl` field contains your GitHub repository's link, you can use the following command when building locally to ignore that value and serve the site on `localhost:4000/`:
+
+     ```shell
+     $ bundle exec jekyll serve --baseurl=""
```
Comment (heiskr) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll.md:62]:
```suggestion
     bundle exec jekyll serve --baseurl=""
```

**Review by nguyenalex836 — DISMISSED** [review]

**Review by nguyenalex836 — APPROVED** [review]

**Comment by nguyenalex836** [comment]
@g-nitin Thank you for committing the suggestion! 💛 I'll update the branch and get this merged once tests are passing 🍏

**Comment by seikh-web** [comment]
### **__**

**Pull request merged**